packages/@webex/plugin-meetings/src/multistream/remoteMediaGroup.ts

```diff
@@ -22,9 +22,10 @@ export class RemoteMediaGroup {
 
   private unpinnedRemoteMedia: RemoteMedia[];
 
-  private mediaRequestId?: MediaRequestId; // id of the "active-speaker" media request id
+  private mediaRequestIds: MediaRequestId[]; // id of the "active-speaker" media request id
 
   private pinnedRemoteMedia: RemoteMedia[];
+  private namedMedia: RemoteMedia[];
```

Comment:
hmm... I don't know... it doesn't feel great having unpinned, pinned and named media arrays all together in this class. The relationship between them is not clear. I see you're splitting the `unpinned` into `named` and `unpinned` in the constructor, but what if someone calls setNamedMediaGroup() on a RemoteMedia object that's in unpinned array later? or what if someone calls `RemoteMediaGroup.pin()` on a named one? This can easily get very messy. Currently named ones are only applicable to audio, right? and pinning/unpinning only really makes sense for video, so maybe we need to create a class hierarchy with AudioRemoteMediaGroup that will support named media groups but no pinning and VideoRemoteMediaGroup that has pinning but no named media groups and both classes would extend RemoteMediaGroup which would be an abstract class now. Unless we can have named media groups for video too? then we would need to do it differently, if that's the case, let's have a call to discuss this.

Reply:
According to the definition of the namedMediaGroup, I think we cannot suppose it's only for audio.  I use the **named** array just for simplifing the logic when creating meida request. I can remove the array for your concern.